src/main/java/uk/nhs/hee/tis/trainee/sync/facade/PlacementEnricherFacade.java

```diff
@@ -371,6 +545,47 @@ private String getTrustName(Trust trust) {
     return trust.getData().get(TRUST_NAME);
   }
 
+  /**
+   * Get the placement for the given id, if the placement is not found it will be requested.
+   *
+   * @param placementId The id of the placement to get.
+   * @return The placement, or Optional.empty() if the ID is null or the placement is not found.
+   */
+  private Optional<Placement> getPlacement(@Nullable String placementId) {
+    if (placementId == null) {
+      return Optional.empty();
+    }
+
+    Optional<Placement> optionalPlacement = placementService.findById(placementId);
+
+    if (optionalPlacement.isEmpty()) {
+      placementService.request(placementId);
+    }
+
+    return optionalPlacement;
+  }
+
+
+  /**
+   * Get the specialty for the given id, if the specialty is not found it will be requested.
+   *
+   * @param specialtyId The id of the specialty to get.
+   * @return The specialty, or Optional.empty() if the ID is null or the specialty is not found.
+   */
+  private Optional<Specialty> getSpecialty(@Nullable String specialtyId) {
```

Comment:
I'd suggest calling this method on L#436 `enrichPlacementWithRelatedSpecialty()` rather than have several places where a specialtyService is called to `get()` or `request()`...
https://github.com/Health-Education-England/tis-trainee-sync/pull/89/files#diff-4f1b2b6f73b355e5c25b3535829dae0e00d480b908ca2d9acca66b8bb7132450R436-R443